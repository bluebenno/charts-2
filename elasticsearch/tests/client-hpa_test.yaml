suite: test client horizontal pod autoscaler
templates:
  - client-hpa.yaml
tests:
  - it: has defaults
    asserts:
      - hasDocuments: { count: 1 }
      - isAPIVersion: { of: autoscaling/v1 }
      - isKind: { of: HorizontalPodAutoscaler }
      - matchSnapshot: { path: spec }
      - equal:
          path: spec.scaleTargetRef
          value:
            apiVersion: apps/v1
            kind: Deployment
            name: RELEASE-NAME-elasticsearch-client
      - equal:
          path: spec.minReplicas
          value: 1
      - equal:
          path: spec.maxReplicas
          value: 2
      - equal:
          path: spec.targetCPUUtilizationPercentage
          value: 70

  - it: can set min replicas
    set:
      client.replicaCount: 3
    asserts:
      - equal:
          path: spec.minReplicas
          value: 3

  - it: can set max replicas
    set:
      client.replicaCount: 3
      client.replicaScaleFactor: 2
    asserts:
      - equal:
          path: spec.maxReplicas
          value: 6

  - it: can set cpu threshold
    set:
      client.cpuThreshold: 90
    asserts:
      - equal:
          path: spec.targetCPUUtilizationPercentage
          value: 90
