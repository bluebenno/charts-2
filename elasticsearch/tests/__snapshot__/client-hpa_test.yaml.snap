has defaults:
  1: |
    maxReplicas: 2
    minReplicas: 1
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: RELEASE-NAME-elasticsearch-client
    targetCPUUtilizationPercentage: 70
